### DISCUSSION TITLE
Dagster equivalent of Airflow's all_done trigger? Or else a better way for dbt assets to check dependencies?

### DISCUSSION THREAD

**Discussion by nmajumdar** [dagster-io/dagster#18256] [Q&A] [answered]
Does Dagster have a way to set an asset to still run even if its upstream dependencies had an error? For instance, if there is Asset A -> Asset B in a job, I want Asset B to wait until after Asset A finished running, but still run even if Asset A failed.

This is really important for us because we use the Dagster DBT integration, but we're finding that it makes our jobs very very fragile. Because of the nature of a handful of dbt ops having many many upstream and downstream dependencies, it means that both:
1. A single upstream dependency failing will prevent **any** dbt models from running, even ones that are completely unrelated to the asset that failed
2. A single error in a dbt model will prevent **any** downstream assets from running, again for downstream assets that aren't related to the model that failed in dbt 

These two together make a single failure almost anywhere in the pipeline likely to make the whole pipeline break. Does anyone know a good way around this problem?

**Comment by benpankow** [comment, marked as answer]
Hi @nmajumdar, right now we don't have a way to mark an upstream dependency as not necessary to run a downstream dependency. We're tracking that issue in this discussion https://github.com/dagster-io/dagster/discussions/16166 & this GitHub issue https://github.com/dagster-io/dagster/issues/15761.

There are some ways to approximate this imperfectly, one solution being to use a sensor to listen for runs of a job to complete (regardless of success or failure) in order to kick off a downstream job.

**Reply by nmaj-ys** [reply]
Thanks for the response! Glad to hear that non-required dependencies is ongoing

Is there any work done on improving how dbt runs in Dagster? If dbt wasn't grouped into so few ops, I wouldn't need to worry about the dependency requirements so much. I would love to see it more intelligently figure out that, evenif upstream Asset A fails, than dbt table C can still run because it doesn't depend on A. In fact, I see it as almost a key feature
